Assets/Plugins/iOS/Shopify/UnityMessageResponder.mm

```diff
@@ -28,13 +28,8 @@
 #import "ProductName-Swift.h"
 
 extern "C" {
-    void _UnityRespondToMessage(const char *identifier, const char *message) {
-        UnityMessageResponse *response = [UnityMessageSender.pendingMessageResponses objectForKey:[NSString stringWithUTF8String:identifier]];
-        
-        if (message != nil) {
-            response.result = [NSString stringWithUTF8String:message];
-        }
-        
-        dispatch_semaphore_signal(response.completion);
+    void _UnityRespondToMessage(const char *identifier, const char *response) {
+        UnityMessage *message = [MessageCenter.messages objectForKey:[NSString stringWithUTF8String:identifier]];
```

Comment:
Would it be better if it was all encapsulated in `MessageCenter` then? 

`[MessageCenter respondToMessageHavingIdentifier: identifier, withResult: response]`